ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/interceptor/InterceptorPipeline.java

```diff
@@ -0,0 +1,331 @@
+                                                   TargetResource targetResource) {
+        if (Objects.isNull(interceptor)) {
+            updateReturnTypeForTarget(infoFromInterceptors, targetResource);
+            return;
+        }
+
+        if (interceptor.isContinueExecution() || !interceptor.isInvokable(targetResource)) {
+            updateReturnTypeForReqInterceptor(interceptor.getNextInReqPath(), infoFromInterceptors, targetResource);
+            if (!interceptor.isInvokable(targetResource)) {
+                return;
+            }
+        }
+
+        if (interceptor.hasErrorReturn()) {
+            updateErrorReturnTypeInReqPath(interceptor, infoFromInterceptors, targetResource);
+        }
+
+        TypeSymbol nonErrorReturnType = interceptor.getNonErrorReturnType();
+        updateNonErrorReturnTypeForInterceptor(interceptor, nonErrorReturnType, infoFromInterceptors, false);
+    }
+
+    private void updateErrorReturnTypeInReqPath(Interceptor interceptor, InfoFromInterceptors infoFromInterceptors,
+                                                TargetResource targetResource) {
+        Interceptor nextErrorInterceptor = interceptor.getNextInReqErrorPath();
+        if (Objects.nonNull(nextErrorInterceptor) &&
+                nextErrorInterceptor.isInvokable(targetResource)) {
+            updateReturnTypeForReqInterceptor(nextErrorInterceptor, infoFromInterceptors, targetResource);
+        } else {
+            nextErrorInterceptor = getNextErrorInterceptor();
+            updateErrorReturnTypeForInterceptor(interceptor, infoFromInterceptors, nextErrorInterceptor);
+        }
+    }
+
+    private Interceptor getNextErrorInterceptor() {
+        Interceptor nextErrorInterceptor;
+        nextErrorInterceptor = null;
+        if (Objects.nonNull(initResInterceptor)) {
+            if (initResInterceptor.getType().equals(InterceptorType.RESPONSE_ERROR)) {
+                nextErrorInterceptor = initResInterceptor;
+            } else {
+                nextErrorInterceptor = initResInterceptor.getNextInResErrorPath();
+            }
+        }
+        return nextErrorInterceptor;
+    }
+
+    private void updateErrorReturnTypeForInterceptor(Interceptor interceptor, InfoFromInterceptors infoFromInterceptors,
+                                                     Interceptor nextErrorInterceptor) {
+        if (Objects.nonNull(nextErrorInterceptor)) {
+            updateReturnTypeForResInterceptor(nextErrorInterceptor, infoFromInterceptors, null, false);
+        } else {
+            infoFromInterceptors.addReturnTypeFromInterceptors(interceptor.getErrorReturnType());
+        }
+    }
+
+    private void updateNonErrorReturnTypeForInterceptor(Interceptor interceptor, TypeSymbol nonErrorReturnType,
+                                                        InfoFromInterceptors infoFromInterceptors, boolean fromTarget) {
+        if (Objects.nonNull(nonErrorReturnType)) {
```

Comment:
```suggestion
        if (nonErrorReturnType != null) {
```

Reply:
For having some readability to code

Reply:
Previously in some other PR it was suggested that using `Objects.nonNull` is more readable. That's why I am using `Objects.isNull` and `Objects.nonNull` throughout the code